### ISSUE TITLE
Change size of pieChart

### ISSUE THREAD

**Issue by JosephOR** [novus/nvd3#1096]
howdy,
I am unable to change the size of the pieChart without changing the code in to main js file. I found an example here https://github.com/novus/nvd3/blob/master/examples/pieChart.html that has width and height variables, but when I change these variables nothing happens, the charts are always the same size. if i remove the !important css, when i hover the slice shrinks, if i remove the interval update, the chart is very small. how can I place a pieChart in a div and fill that div whatever the size is without changing the main js file.
thanks, joey

**Comment by robinfhu** [comment]
Do you need the pie chart to be responsive to the browser width?

**Comment by liquidpele** [comment]
duplicate of https://github.com/novus/nvd3/issues/810 and https://github.com/novus/nvd3/issues/937 I think...

**Comment by JosephOR** [comment]
Hi Robinfhu,
The problem is the 'g' element inside the svg, ( class = nvd3 nv-wrap nv-pieChart) is not filling the svg div. I tried doing css, which i can see is appended in the inspect element tool, but it does not change any properties. it can be done editing the js file but I don't want to do that as I am loading the newest version automatically. another problem i'm having is I would like to be able to toggle the legend(show/hide) on a button click. is it possible to do this without reloading the chart again. 
Thanks

**Comment by JosephOR** [comment]
Oh and sorry I don't need it to be responsive, its a static width site.

**Comment by robinfhu** [comment]
Reloading the chart is necessary to change its size and also to remove the legend. Looking at the code, it looks like the width and height are fixed. Changing the chart properties and calling `chart.update` should do it.  

`chart.showLegend(false).update()`

**Comment by robinfhu** [comment]
My bad, I don't think the above code will actually hide the legend. You maybe have to use CSS to toggle it

`.nv-legendWrap { display: none;}`

**Comment by JosephOR** [comment]
Yeah I had tried the first option already, taught it would work. yeah the css works, taught there might have been an nvd3 way. cheers for the help

**Comment by danideng** [comment]
This works! You must contain <nvd3> inside a <div> then set the div's max-hight less than the height in the translate:
`<div style="max-height:200px; max-width:250px; position:relative;">
     <nvd3 [options]="pieoptions" [data]="piedata"></nvd3> 
  </div>`
In my component.ts, pieheight is set to 250.
